src/main/groovy/org/grails/gradle/plugin/publishing/GrailsPublishGradlePlugin.groovy

```diff
@@ -111,22 +120,46 @@ The credentials and connection url must be specified as a project property or an
+        String detectedVersion = (project.version == Project.DEFAULT_VERSION ? (project.findProperty('projectVersion') ?: Project.DEFAULT_VERSION) : project.version) as String
+        if (detectedVersion == Project.DEFAULT_VERSION) {
+            throw new IllegalStateException("Project `${project.name}` has an unspecified version (neither `version` or the property `projectVersion` is defined). Release state cannot be determined.")
+        }
+        project.rootProject.logger.info("Detected Version $detectedVersion for Project `${project.name}`")
+
+        boolean isSnapshot = detectedVersion.endsWith('SNAPSHOT')
+        if (isSnapshot) {
+            project.rootProject.logger.info("Snapshot detected for Project `${project.name}`")
+        }
         boolean isRelease = !isSnapshot
-        boolean mavenPublish = (isSnapshot && snapshotPublishType == PublishType.MAVEN_PUBLISH) || (isRelease && releasePublishType == PublishType.MAVEN_PUBLISH)
-        boolean nexusPublish = (isSnapshot && snapshotPublishType == PublishType.NEXUS_PUBLISH) || (isRelease && releasePublishType == PublishType.NEXUS_PUBLISH)
+        if (isRelease) {
+            project.rootProject.logger.info("Release detected for Project `${project.name}`")
+        }
 
-        final PluginManager projectPluginManager = project.getPluginManager()
-        final PluginManager rootProjectPluginManager = project.rootProject.getPluginManager()
+        boolean useMavenPublish = (isSnapshot && snapshotPublishType == PublishType.MAVEN_PUBLISH) || (isRelease && releasePublishType == PublishType.MAVEN_PUBLISH)
+        if (useMavenPublish) {
+            project.rootProject.logger.info("Maven Publish is enabled for Project `${project.name}`")
+        }
+        boolean useNexusPublish = (isSnapshot && snapshotPublishType == PublishType.NEXUS_PUBLISH) || (isRelease && releasePublishType == PublishType.NEXUS_PUBLISH)
+        if (useNexusPublish) {
+            project.rootProject.logger.info("Nexus Publish is enabled for Project `${project.name}`")
+        }
 
         // Required for the pom always
+        final PluginManager projectPluginManager = project.pluginManager
         projectPluginManager.apply(MavenPublishPlugin)
 
-        if (nexusPublish) {
+        if (useNexusPublish) {
+            // The nexus plugin is special since it must always be applied to the root project.
+            // Handle when multiple subprojects exist and grailsPublish is defined in each one instead of at the root.
+            final PluginManager rootProjectPluginManager = project.rootProject.pluginManager
+            boolean hasNexusPublishApplied = rootProjectPluginManager.hasPlugin(NEXUS_PUBLISH_PLUGIN_ID)
+            if (hasNexusPublishApplied) {
+                project.rootProject.logger.info("Nexus Publish Plugin already applied to root project")
+            }
+
             rootProjectPluginManager.apply(NexusPublishPlugin)
```

Comment:
I tried that & I was hoping it would fix the Geb project, but it doesn't.  

This actually works when you have a project structure so that there is a common root build.gradle and several subprojects with build.gradle in them.  You can reapply it without issue in my testing.  It wouldn't hurt to have it an else, so I'm going to add it there.